File: 1028-interval-list-intersections/1028-interval-list-intersections.py
class Solution:
    def intervalIntersection(self, fl: List[List[int]], sl: List[List[int]]) -> List[List[int]]:
        res = []
        l1 = 0
        l2 = 0
        while len(fl) > l1 and len(sl) > l2:
            start = max(fl[l1][0], sl[l2][0])
            end = min(fl[l1][1], sl[l2][1])
            if start <= end:
                res.append([start, end])
            if fl[l1][1] == end:
                l1 += 1
            else:
                l2 += 1
        return res

File: 1028-interval-list-intersections/1028-interval-list-intersections.java
class Solution {
    public int[][] intervalIntersection(int[][] f, int[][] s) {
        List<int[]> arr = new ArrayList<>();
        int p1 = 0, p2 = 0;
        while (p1 < f.length && p2 < s.length) {
            int start = Math.max(f[p1][0], s[p2][0]);
            int end = Math.min(f[p1][1], s[p2][1]);
            if (start <= end) {
                arr.add(new int[] {start, end});
            }
            if (f[p1][1] < s[p2][1]) {
                p1++;
            } else {
                p2++;
            }
        }
        int[][] res = new int[arr.size()][2];
        for (int i = 0; i < arr.size(); i++) {
            res[i] = arr.get(i);
        }
        return res;
    }
}